typing/typecore.ml

```diff
@@ -4222,6 +4219,15 @@ let rec is_nonexpansive exp =
      which is non-expansive: *)
   | Texp_hole _ -> true
 
+and is_nonexpansive_prim (prim : Primitive.description) args =
+  match prim.prim_name, args with
+  | ("%raise" | "%reraise" | "%raise_notrace"), [Nolabel, Arg (e, _)] ->
+    is_nonexpansive e
+  | ("%identity" | "%obj_magic"), [Nolabel, Arg (e, _)]
+    when not (Language_extension.erasable_extensions_only ()) ->
```

Comment:
Maybe add a test demonstrating how the behavior of this depends on the extensions choice?

Reply:
I've pushed this test.

Reply:
Thanks!